src/rust/engine/options/src/args.rs

```diff
@@ -97,20 +97,25 @@ impl Arg {
 #[derive(Debug)]
 pub struct Args {
     args: Vec<Arg>,
+    #[allow(dead_code)]
+    pub passthrough_args: Option<Vec<String>>,
```

Comment:
Is there a semantic difference between an empty list and a None? I tend to find this is an anti-pattern in most code, and I can't see why this code would care if we did `pants run foo` vs `pants run foo --`... in both cases there's no args.

I'm also a strong proponent for never mixing visibility specifiers, https://rust-lang.github.io/rust-clippy/master/index.html#/partial_pub_fields. I find it leads to easier-to-change code. In this case, it's also an artifact of our weird test structure; which isn't very common in Rust. If `args_tests` was nested within `args` it'd not need to be public (yet?). In which case it'd probably make more sense to think about an accessor when it gets actual use.

Reply:
There could be, depending on how passthrough args are consumed. For example, if the logic is that passthrough args override pants-generated args, then this is the way to say "override with empty args".

Reply:
Re visibility specifiers, yeah, perhaps an accessor is a good idea. We definitely don't want to make the `args` field public.

Possibly we should shake up the test structure?